dbt_platform_helper/providers/cache.py

```diff
@@ -0,0 +1,76 @@
+import os
+from datetime import datetime
+from pathlib import Path
+
+import yaml
+
+
+class CacheProvider:
+    def __init__(self):
+        self.cache_file = ".platform-helper-config-cache.yml"
+
+    def read_supported_versions_from_cache(self, resource_name):
```

Comment:
Maybe generalise this a bit more?
```suggestion
    def read_from_cache(self, resource_name):
```
Might be used for somethig other than supported versions.

Reply:
atm its specific to versions:

`return platform_helper_config.get(resource_name).get("versions")`, I expect that not every object we cache will follow the same structure so it probably makes more sense to leave this as a specific method.

Reply:
Ah, yes. Good for now.